syscalls.go

```diff
@@ -244,3 +244,32 @@ var haveProbeReadKernel = internal.FeatureTest("bpf_probe_read_kernel", "5.5", f
 	_ = fd.Close()
 	return nil
 })
+
+var haveProgAttachType = internal.FeatureTest("expected_attach_type", "4.17", func() error {
```

Comment:
This needs a corresponding call to `testutils.CheckFeatureTest` or whatever it's called.

Reply:
Good catch, added `TestHaveProgAttachType`.